uisdk/src/main/java/com/karhoo/uisdk/screen/rides/upcoming/card/UpcomingRideCardPresenter.kt

```diff
@@ -43,4 +53,19 @@ class UpcomingRideCardPresenter(view: UpcomingRideCardMVP.View,
         view?.goToDetails(trip)
     }
 
+    private fun checkCancellationSLAMinutes(tripStatus: TripStatus, serviceCancellation: ServiceCancellation?, context: Context) {
+        if (serviceCancellation?.hasValidCancellationDependingOnTripStatus(tripStatus) == false) {
+            view?.showCancellationText(false)
+            return
+        }
+
+        val text = serviceCancellation?.getCancellationText(context)
```

Comment:
I'm wondering if we should be getting the Strings in the Presenter or can it be coded in such a way that a different method is called on the View? @mohadian any thoughts?

Reply:
Using the extension is not a bad idea, it would keep the logic in one place and avoid repeating the logic. The other alternative is to use a utility class but I think Kotlin favors extensions to utility classes

Reply:
Seems better to avoid the Context in the Presenter but then guess it means that you end up with the same logic in a couple of places. Anyways, if you're both happy with it, then go for it.

Reply:
Having said that, the logic would still remain in the extension. It would just be called from the View instead.

Reply:
Anyways, have approved :)